swatch-metrics/src/main/java/com/redhat/swatch/metrics/service/prometheus/PrometheusService.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.redhat.swatch.metrics.service.prometheus;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.redhat.swatch.clients.prometheus.api.model.QueryResultDataResultInner;
+import com.redhat.swatch.clients.prometheus.api.model.ResultType;
+import com.redhat.swatch.clients.prometheus.api.model.StatusType;
+import com.redhat.swatch.clients.prometheus.api.resources.ApiException;
+import com.redhat.swatch.clients.prometheus.api.resources.QueryApi;
+import com.redhat.swatch.clients.prometheus.api.resources.QueryRangeApi;
+import com.redhat.swatch.metrics.exception.ErrorCode;
+import com.redhat.swatch.metrics.exception.ExternalServiceException;
+import com.redhat.swatch.metrics.service.prometheus.model.QuerySummaryResult;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import java.io.File;
+import java.io.IOException;
+import java.time.OffsetDateTime;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Wraps prometheus specific API calls to make them more application specific. */
+@ApplicationScoped
+public class PrometheusService {
+
+  private static final Logger log = LoggerFactory.getLogger(PrometheusService.class);
+  private static final String JSON_PROPERTY_RESULT_TYPE = "resultType";
+  private static final String JSON_PROPERTY_RESULT = "result";
+  public static final String JSON_PROPERTY_STATUS = "status";
+  public static final String JSON_PROPERTY_DATA = "data";
+  public static final String JSON_PROPERTY_ERROR_TYPE = "errorType";
+  public static final String JSON_PROPERTY_ERROR = "error";
+
+  @Inject @RestClient QueryApi queryApi;
+  @Inject @RestClient QueryRangeApi queryRangeApi;
+  @Inject JsonFactory factory;
+
+  public QuerySummaryResult runRangeQuery(
+      String query,
```

Comment:
We previously had to manually URL-escape the query. Is this no longer needed :-) ?

Reply:
No, this is one of the two things I spotted when testing the changes (see the Testing section in the description).